sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -847,6 +847,11 @@ object SQLConf {
       .intConf
       .createWithDefault(UnsafeExternalSorter.DEFAULT_NUM_ELEMENTS_FOR_SPILL_THRESHOLD.toInt)
 
+  val SUPPORT_QUOTED_REGEX_COLUMN_NAME = buildConf("spark.sql.parser.quotedRegexColumnNames")
+    .doc("When true, a SELECT statement can take regex-based column specification.")
```

Comment:
We also need to explain the impact in the description. For example,
> When true, quoted Identifiers (using backticks) are interpreted as regular expressions. 

Feel free to input the above text